--- evolution-data-server-1.0.2/src/server.c.workaround-cal-backend-leak	2004-10-20 01:11:58.049726116 -0400
+++ evolution-data-server-1.0.2/src/server.c	2004-10-20 01:13:53.582289628 -0400
@@ -151,6 +151,13 @@
 		bonobo_main_quit ();
 	}
 
+	/* Workaround for http://bugzilla.ximian.com/show_bug.cgi?id=68533 */
+	if (e_data_cal_factory_get_n_backends (e_data_cal_factory) == 0 &&
+	    e_data_book_factory_get_n_backends (e_data_book_factory) == 1) {
+		g_message ("termination_handler(): Terminating the Server (with workaround for bug 68533).  Have a nice day.");
+		bonobo_main_quit ();
+	}
+
 	termination_handler_id = 0;
 	return FALSE;
 }
