diff -aurN evolution-data-server-1.7.91.orig/libedataserverui/Makefile.am evolution-data-server-1.7.91/libedataserverui/Makefile.am
--- evolution-data-server-1.7.91.orig/libedataserverui/Makefile.am	2006-08-07 06:48:26.000000000 +0200
+++ evolution-data-server-1.7.91/libedataserverui/Makefile.am	2006-08-07 22:44:08.109276133 +0200
@@ -42,7 +42,7 @@
 else
 libedataserverui_1_2_la_LIBADD = 				\
 	$(top_builddir)/addressbook/libebook/libebook-1.2.la	\
-	$(E_DATA_SERVER_LIBS)
+	$(E_DATA_SERVER_UI_LIBS)
 endif
 
 libedataserverui_1_2_la_LDFLAGS = 										\
