diff -aurN evolution-data-server-1.5.5.orig/Makefile.am evolution-data-server-1.5.5/Makefile.am
--- evolution-data-server-1.5.5.orig/Makefile.am	2005-12-08 08:54:12.000000000 +0100
+++ evolution-data-server-1.5.5/Makefile.am	2006-01-22 21:07:02.287051000 +0100
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = 1.6
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = libdb win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
+SUBDIRS = win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
 DIST_SUBDIRS= win32 libedataserver servers camel addressbook calendar libedataserverui src docs art po
 
 changelogs =			\
@@ -27,9 +27,10 @@
 	intltool-update.in				\
 	intltool-extract.in				\
 	evolution-data-server-zip.in			\
-	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in) 	\
-	$(LIBDB_FILES)
+	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in)
 
+# files below should be removed due to system db patch, but they are left
+# so patch is small
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update iconv-detect.h $(pkgconfig_DATA)
 distclean-local:
 	(cd libdb && $(MAKE) $(AM_MAKEFLAGS) distclean)
diff -aurN evolution-data-server-1.5.5.orig/addressbook/backends/file/Makefile.am evolution-data-server-1.5.5/addressbook/backends/file/Makefile.am
--- evolution-data-server-1.5.5.orig/addressbook/backends/file/Makefile.am	2005-06-22 14:34:21.000000000 +0200
+++ evolution-data-server-1.5.5/addressbook/backends/file/Makefile.am	2006-01-22 21:02:28.917051000 +0100
@@ -4,7 +4,6 @@
 	-I$(top_builddir)				\
 	-I$(top_srcdir)/addressbook			\
 	-I$(top_builddir)/addressbook			\
-	-I$(top_srcdir)/libdb/dist			\
         $(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
 extension_LTLIBRARIES = libebookbackendfile.la
@@ -18,8 +17,8 @@
 	$(top_builddir)/addressbook/libebook/libebook-1.2.la		\
 	$(top_builddir)/addressbook/libedata-book/libedata-book-1.2.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la		\
-	$(top_builddir)/libdb/dist/libdb-4.1.la				\
-	$(EVOLUTION_ADDRESSBOOK_LIBS)
+	$(EVOLUTION_ADDRESSBOOK_LIBS)					\
+	-ldb
 
 libebookbackendfile_la_LDFLAGS =	\
 	-module -avoid-version $(NO_UNDEFINED)
diff -aurN evolution-data-server-1.5.5.orig/configure.in evolution-data-server-1.5.5/configure.in
--- evolution-data-server-1.5.5.orig/configure.in	2006-01-17 09:48:52.000000000 +0100
+++ evolution-data-server-1.5.5/configure.in	2006-01-22 21:03:31.387051000 +0100
@@ -1270,7 +1270,6 @@
 export privincludedir
 export privdatadir
 AC_CONFIG_SUBDIRS(calendar/libical)
-AC_CONFIG_SUBDIRS(libdb/dist)
 
 AC_OUTPUT([
 Makefile
@@ -1328,7 +1327,6 @@
 camel/tests/stream/Makefile
 camel/camel.pc
 camel/camel-provider.pc
-libdb/Makefile
 libedataserver/Makefile
 libedataserver/libedataserver.pc
 libedataserverui/Makefile
diff -aurN evolution-data-server-1.5.5.orig/libedataserver/Makefile.am evolution-data-server-1.5.5/libedataserver/Makefile.am
--- evolution-data-server-1.5.5.orig/libedataserver/Makefile.am	2006-01-16 09:43:14.000000000 +0100
+++ evolution-data-server-1.5.5/libedataserver/Makefile.am	2006-01-22 21:05:44.877051000 +0100
@@ -1,7 +1,6 @@
 INCLUDES =								\
 	-I$(top_srcdir)							\
 	-DG_LOG_DOMAIN=\"e-data-server\"				\
-	-I$(top_srcdir)/libdb/dist					\
 	-DE_DATA_SERVER_PREFIX=\"$(prefix)\"				\
 	-DEVOLUTION_LOCALEDIR=\""$(localedir)"\"			\
 	-DE_DATA_SERVER_EXTENSIONDIR=\"$(extensiondir)\"		\
@@ -49,7 +48,7 @@
 libedataserver_1_2_la_LIBADD = 				\
 	$(E_DATA_SERVER_LIBS)				\
 	$(ICONV_LIBS)					\
-	$(top_builddir)/libdb/dist/libdb-4.1.la		\
+	-ldb						\
 	$(SOCKET_LIBS)
 
 libedataserver_1_2_la_LDFLAGS = \
