diff -Nur evolution-data-server-1.7.90.1.orig/libedataserverui/Makefile.am evolution-data-server-1.7.90.1.as_needed-fix/libedataserverui/Makefile.am
--- evolution-data-server-1.7.90.1.orig/libedataserverui/Makefile.am	2006-07-22 10:59:09.000000000 +0000
+++ evolution-data-server-1.7.90.1.as_needed-fix/libedataserverui/Makefile.am	2006-08-06 18:24:42.000000000 +0000
@@ -36,7 +36,7 @@
 
 libedataserverui_1_2_la_LIBADD = 				\
 	$(top_builddir)/addressbook/libebook/libebook-1.2.la	\
-	$(E_DATA_SERVER_LIBS)
+	$(E_DATA_SERVER_UI_LIBS)
 
 libedataserverui_1_2_la_LDFLAGS = 										\
 	-version-info $(LIBEDATASERVERUI_CURRENT):$(LIBEDATASERVERUI_REVISION):$(LIBEDATASERVERUI_AGE) 
