diff -urN evolution-data-server-0.0.93.orig/configure.in evolution-data-server-0.0.93/configure.in
--- evolution-data-server-0.0.93.orig/configure.in	2004-05-22 12:08:38.934333552 +0200
+++ evolution-data-server-0.0.93/configure.in	2004-05-22 12:15:55.995890096 +0200
@@ -88,7 +88,7 @@
 dnl I18N stuff
 AC_PROG_INTLTOOL
 
-ALL_LINGUAS="am az be bg ca cs da de el en_AU en_CA en_GB es et eu fi fr ga gl hr hu it ja ko lt lv mn ms nl nn no pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="am az be bg ca cs da de el en_AU en_CA en_GB es et eu fi fr ga gl hr hu it ja ko lt lv mn ms nb nl nn pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 GETTEXT_PACKAGE=evolution-data-server-1.5
