diff -urN evolution-data-server-0.0.98.orig/configure.in evolution-data-server-0.0.98/configure.in
--- evolution-data-server-0.0.98.orig/configure.in	2004-08-15 11:49:09.594192952 +0200
+++ evolution-data-server-0.0.98/configure.in	2004-08-15 11:49:44.443894992 +0200
@@ -88,7 +88,7 @@
 dnl I18N stuff
 AC_PROG_INTLTOOL
 
-ALL_LINGUAS="am az be bg bn ca cs da de el en_AU en_CA en_GB es et eu fi fr ga gl gu hr hu it ja ko lt lv mn ms nl nn no pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi zh_CN zh_TW"
+ALL_LINGUAS="am az be bg bn ca cs da de el en_AU en_CA en_GB es et eu fi fr ga gl gu hr hu it ja ko lt lv mn ms nb nl nn pa pl pt pt_BR ro ru sk sl sq sr sr@Latn sv tr uk vi zh_CN zh_TW"
 AM_GLIB_GNU_GETTEXT
 
 GETTEXT_PACKAGE=evolution-data-server-1.5
