diff -aurN evolution-data-server-1.3.4.orig/addressbook/backends/file/Makefile.am evolution-data-server-1.3.4/addressbook/backends/file/Makefile.am
--- evolution-data-server-1.3.4.orig/addressbook/backends/file/Makefile.am	2005-06-25 12:17:17.000000000 +0200
+++ evolution-data-server-1.3.4/addressbook/backends/file/Makefile.am	2005-07-03 18:26:24.000000000 +0200
@@ -4,7 +4,6 @@
 	-I$(top_builddir)				\
 	-I$(top_srcdir)/addressbook			\
 	-I$(top_builddir)/addressbook			\
-	-I$(top_srcdir)/libdb/dist			\
         $(EVOLUTION_ADDRESSBOOK_CFLAGS)
 
 extension_LTLIBRARIES = libebookbackendfile.la
@@ -18,8 +17,8 @@
 	$(top_builddir)/addressbook/libebook/libebook-1.2.la		\
 	$(top_builddir)/addressbook/libedata-book/libedata-book-1.2.la	\
 	$(top_builddir)/libedataserver/libedataserver-1.2.la		\
-	$(top_builddir)/libdb/dist/libdb-4.1.la				\
-	$(EVOLUTION_ADDRESSBOOK_LIBS)
+	$(EVOLUTION_ADDRESSBOOK_LIBS)					\
+	-ldb
 
 libebookbackendfile_la_LDFLAGS =	\
 	-module -avoid-version $(NO_UNDEFINED)
diff -aurN evolution-data-server-1.3.4.orig/configure.in evolution-data-server-1.3.4/configure.in
--- evolution-data-server-1.3.4.orig/configure.in	2005-07-02 12:09:23.000000000 +0200
+++ evolution-data-server-1.3.4/configure.in	2005-07-03 18:13:40.000000000 +0200
@@ -1213,7 +1213,6 @@
 export privincludedir
 export privdatadir
 AC_CONFIG_SUBDIRS(calendar/libical)
-AC_CONFIG_SUBDIRS(libdb/dist)
 
 AC_OUTPUT([
 Makefile
@@ -1268,7 +1267,6 @@
 camel/tests/stream/Makefile
 camel/camel.pc
 camel/camel-provider.pc
-libdb/Makefile
 libedataserver/Makefile
 libedataserver/libedataserver.pc
 libedataserverui/Makefile
diff -aurN evolution-data-server-1.3.4.orig/libedataserver/Makefile.am evolution-data-server-1.3.4/libedataserver/Makefile.am
--- evolution-data-server-1.3.4.orig/libedataserver/Makefile.am	2005-04-11 14:21:32.000000000 +0200
+++ evolution-data-server-1.3.4/libedataserver/Makefile.am	2005-07-03 18:14:15.000000000 +0200
@@ -1,7 +1,6 @@
 INCLUDES =								\
 	-I$(top_srcdir)							\
 	-DG_LOG_DOMAIN=\"e-data-server\"				\
-	-I$(top_srcdir)/libdb/dist					\
 	-DE_DATA_SERVER_EXTENSIONDIR=\"$(extensiondir)\"		\
 	-DE_DATA_SERVER_IMAGESDIR=\"$(imagesdir)\"			\
 	$(E_DATA_SERVER_CFLAGS)
@@ -44,8 +43,8 @@
 libedataserver_1_2_la_LIBADD = 				\
 	$(E_DATA_SERVER_LIBS)				\
 	$(ICONV_LIBS)					\
-	$(top_builddir)/libdb/dist/libdb-4.1.la
-
+	-ldb
+	
 libedataserver_1_2_la_LDFLAGS = 										\
 	-version-info $(LIBEDATASERVER_CURRENT):$(LIBEDATASERVER_REVISION):$(LIBEDATASERVER_AGE)
 
diff -aurN evolution-data-server-1.3.4.orig/Makefile.am evolution-data-server-1.3.4/Makefile.am
--- evolution-data-server-1.3.4.orig/Makefile.am	2005-04-11 14:21:26.000000000 +0200
+++ evolution-data-server-1.3.4/Makefile.am	2005-07-03 18:18:54.000000000 +0200
@@ -1,7 +1,7 @@
 AUTOMAKE_OPTIONS = 1.6
 DISTCHECK_CONFIGURE_FLAGS = --enable-gtk-doc
 
-SUBDIRS = libdb libedataserver servers camel addressbook calendar libedataserverui src docs art po
+SUBDIRS = libedataserver servers camel addressbook calendar libedataserverui src docs art po
 DIST_SUBDIRS= libedataserver servers camel addressbook calendar libedataserverui src docs art po
 
 changelogs =			\
@@ -26,9 +26,10 @@
 	intltool-merge.in				\
 	intltool-update.in				\
 	intltool-extract.in				\
-	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in) 	\
-	$(LIBDB_FILES)
+	$(pkgconfig_DATA:-$(API_VERSION).pc=.pc.in)
 
+# files below should be removed due to system db patch, but they are left,
+# so patch is small
 DISTCLEANFILES = intltool-extract intltool-merge intltool-update iconv-detect.h $(pkgconfig_DATA)
 distclean-local:
 	(cd libdb && $(MAKE) $(AM_MAKEFLAGS) distclean)
