diff -urN aa/evolution-data-server-0.0.97/calendar/libical/configure.in evolution-data-server-0.0.97/calendar/libical/configure.in
--- aa/evolution-data-server-0.0.97/calendar/libical/configure.in	2004-02-26 21:25:50.000000000 +0100
+++ evolution-data-server-0.0.97/calendar/libical/configure.in	2004-08-03 15:01:03.538422040 +0200
@@ -2,13 +2,13 @@
 AC_INIT(src)
 AC_CANONICAL_SYSTEM
 
-AM_INIT_AUTOMAKE(libical,0.24) 
-AM_CONFIG_HEADER(config.h)
-
 dnl Set the aux dir to .. (the toplevel evolution directory) so ylwrap
 dnl is found in the correct directory for automake >= 1.5
 AC_CONFIG_AUX_DIR(..)
 
+AM_INIT_AUTOMAKE(libical,0.24) 
+AM_CONFIG_HEADER(config.h)
+
 dnl Subst in evolution's private directories
 AC_SUBST(privlibdir)
 AC_SUBST(privincludedir)
