diff -urN aaa/evolution-data-server-1.7.91/configure.in evolution-data-server-1.7.91/configure.in
--- aaa/evolution-data-server-1.7.91/configure.in	2006-08-07 19:04:24.000000000 +0200
+++ evolution-data-server-1.7.91/configure.in	2006-08-13 23:43:09.773575500 +0200
@@ -561,7 +561,7 @@
 			AC_CACHE_CHECK([for Mozilla nss libraries], moz_nss_libs,
 			[
 	 			LIBS_save="$LIBS"
-				LDFLAGS="$LDFLAGS -L$with_nspr_libs $nsprlibs -L$with_nss_libs $nsslibs"
+				LDFLAGS="$LDFLAGS -L$with_nspr_libs $nsprlibs -L$with_nss_libs"
 		      		LIBS="$nsslibs $nsprlibs"
 				AC_TRY_LINK_FUNC(NSS_Init, moz_nss_libs="yes", moz_nss_libs="no")
 				if test "$moz_nss_libs" = no; then
@@ -950,17 +950,21 @@
 msg_krb5="no"
 if test "x${with_krb5}" != "xno"; then
 	LDFLAGS_save="$LDFLAGS"
+	LIBS_save="$LIBS"
 	
 	mitlibs="-lkrb5 -lk5crypto -lcom_err -lgssapi_krb5"
 	heimlibs="-lkrb5 -lcrypto -lasn1 -lcom_err -lroken -lgssapi"
 	AC_CACHE_CHECK([for Kerberos 5], ac_cv_lib_kerberos5,
 	[
-		LDFLAGS="$LDFLAGS -L$with_krb5_libs $mitlibs"
+		LIBS="$LIBS $mitlibs"
+		LDFLAGS="$LDFLAGS -L$with_krb5_libs"
 		AC_TRY_LINK_FUNC(krb5_init_context, ac_cv_lib_kerberos5="$mitlibs",
 		[
-			LDFLAGS="$LDFLAGS_save -L$with_krb5_libs $heimlibs"
+			LIBS="$LIBS_save $heimlibs"
+			LDFLAGS="$LDFLAGS_save -L$with_krb5_libs"
 			AC_TRY_LINK_FUNC(krb5_init_context, ac_cv_lib_kerberos5="$heimlibs", ac_cv_lib_kerberos5="no")
 		])
+		LIBS="$LIBS_save"
 		LDFLAGS="$LDFLAGS_save"
 	])
 	if test "$ac_cv_lib_kerberos5" != "no"; then
@@ -995,6 +999,7 @@
 
 msg_krb4="no"
 if test "x${with_krb4}" != "xno"; then
+	LIBS_save="$LIBS"
 	LDFLAGS_save="$LDFLAGS"
 	AC_CACHE_CHECK(for Kerberos 4, ac_cv_lib_kerberos4,
 	[
@@ -1002,24 +1007,28 @@
 
 		mitcompatlibs="-lkrb4 -ldes425 -lkrb5 -lk5crypto -lcom_err"
 		# Look for MIT krb5 compat krb4
-		LDFLAGS="$LDFLAGS -L$with_krb4_libs $mitcompatlibs"
+		LIBS="$LIBS $mitcompatlibs"
+		LDFLAGS="$LDFLAGS -L$with_krb4_libs"
 		AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="$mitcompatlibs")
 		
 		if test "$ac_cv_lib_kerberos4" = "no"; then
 			# Look for KTH krb4
-			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs -lkrb -lcrypto -lcom_err -lroken"
+			LIBS="$LIBS_save -lkrb -lcrypto -lcom_err -lroken"
+			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs"
 			AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb -lcrypto -lcom_err -lroken")
 		fi
 		if test "$ac_cv_lib_kerberos4" = "no"; then
 			# Look for old MIT krb4
-			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs -lkrb"
+			LIBS="$LIBS_save -lkrb"
+			LDFLAGS="$LDFLAGS_save -L$with_krb4_libs"
 			AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb",
 			[
-				LDFLAGS="$LDFLAGS -ldes"
+				LIBS="$LIBS -ldes"
 				AC_TRY_LINK_FUNC(krb_mk_req, ac_cv_lib_kerberos4="-lkrb -ldes")
 			])
 		fi
 	])
+	LIBS="$LIBS_save"
 	LDFLAGS="$LDFLAGS_save"
 	if test "$ac_cv_lib_kerberos4" != "no"; then
 		AC_DEFINE(HAVE_KRB4,1,[Define if you have Krb4])
