--- evolution-data-server-1.2.2/calendar/libecal/e-cal.c~	2005-04-08 15:11:41.000000000 +0200
+++ evolution-data-server-1.2.2/calendar/libecal/e-cal.c	2005-04-29 20:41:01.710986876 +0200
@@ -885,6 +885,9 @@
 	g_mutex_unlock (op->mutex);	
 }
 
+static gboolean
+open_calendar (ECal *ecal, gboolean only_if_exists, GError **error, ECalendarStatus *status, gboolean needs_auth);
+
 static gboolean  
 reopen_with_auth (gpointer data)
 {
